source/extensions/stat_sinks/circonus/circonus_stat_view.cc

```diff
@@ -0,0 +1,107 @@
+#include "extensions/stat_sinks/circonus/circonus_stat_view.h"
+
+#include <google/protobuf/util/internal/json_objectwriter.h>
+#include <google/protobuf/io/zero_copy_stream.h>
+#include <iostream>
+
+#include "common/buffer/buffer_impl.h"
+#include "common/config/well_known_names.h"
+#include "common/http/headers.h"
+#include "common/http/utility.h"
+#include "common/stats/utility.h"
+
+#include "circllhist.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace StatSinks {
+namespace Circonus {
+
+namespace {
+const char* METRIC_HISTOGRAM_CUMULATIVE = "H";
+const char* METRIC_HISTOGRAM = "h";
+
+const char* chooseHistogramsType(absl::string_view path_and_query) {
+  const auto qp = Envoy::Http::Utility::parseQueryString(path_and_query);
+  auto found = qp.find("histogram_type");
+
+  if (found != qp.end() && found->second == "cumulative") {
+    return METRIC_HISTOGRAM_CUMULATIVE;
+  }
+
+  return METRIC_HISTOGRAM;
+}
+} // namespace
+
+CirconusStatView::CirconusStatView(Server::Instance& server) : server_(server) {
+  server.admin().addHandler("/stats/circonus", "Display Circonus metrics on the admin console",
+                            MAKE_ADMIN_HANDLER(handlerAdminStats), false, false);
+}
+
+Http::Code CirconusStatView::handlerAdminStats(absl::string_view path_and_query,
+                                               Http::ResponseHeaderMap& response_headers,
+                                               Buffer::Instance& response,
+                                               Server::AdminStream& /* admin_stream */) {
+  response_headers.setContentType(Http::Headers::get().ContentTypeValues.Json);
+  response.add(makeJsonBody(chooseHistogramsType(path_and_query)));
+  return Http::Code::OK;
+}
+
+std::string CirconusStatView::makeJsonBody(const char* histogram_type) {
+  const auto& stats = server_.stats();
+  const auto histograms = stats.histograms();
+  auto& bb = bb_;
```

Comment:
I'm not sure the admin is single-threaded and I don't have time at the moment to puzzle it out. Can you make sure reusing this field is safe?

FWIW, In the prometheus path we simply write directly to `resoibse`. That's probably hard to do here given json output and the call signature of `hist_serialize_b64` so I'm ok with the `std::vector<uint8_t>`.

Reply:
Per the `Hystrix` example, it is.  

> Can you make sure reusing this field is safe?

I did.